core/src/main/java/io/grpc/Metadata.java

```diff
@@ -490,6 +491,7 @@ private boolean bytesEqual(byte[] left, byte[] right) {
   }
 
   /** Marshaller for metadata values that are serialized into raw binary. */
+  @Immutable
```

Comment:
It is good to be defensive, but in this case we really cannot enforce it.  I liken `@Immutable` to a promise, rather than a suggestion.  We can add it to classes we own, but can't enforce it for others.